Kitodo/src/main/java/org/kitodo/production/migration/TemplateComparator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.migration;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+import org.kitodo.data.database.beans.Template;
+
+public class TemplateComparator implements Comparator<Template> {
+
+    @Override
+    public int compare(Template firstTemplate, Template secondTemplate) {
+        if (Objects.isNull(firstTemplate) || Objects.isNull(secondTemplate)) {
+            return 1;
```

Comment:
This `compare` method will return `1` (e.g. "unequal") if both templates are `null`. Is that intended?

Reply:
Yes, null-templates are not allowed in the matching context

Reply:
That may be true, but a check comparing `null` with `null` should normally return `true`, shouldn't it? That the templates aren't `null` - because that's not allowed here - should be checked somewhere else.